### PR TITLE
fix: patchedDependencies works on sub dir and add e2e test

### PR DESCRIPTION
**Pull request by hunshcn**
fix https://github.com/aspect-build/rules_js/issues/975 and add e2e test for patchedDependencies

### CODE REVIEW

**Comment by hunshcn** [comment]
relate to https://github.com/aspect-build/rules_js/pull/832
@kormide

File: e2e/npm_translate_lock_patch/portal/test.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Importing segfault-handler fails with `Error: Could not locate the bindings file.`
+// if `node-gyp rebuild` has not been run.
```
Comment (kormide) [inline e2e/npm_translate_lock_patch/portal/test.js:2]: I think this comment is a copy pasta from another test. Can be removed?

File: e2e/npm_translate_lock_patch/MODULE.bazel [hunk 2]
```diff
@@ -0,0 +1,32 @@
+module(
+    name = "e2e_npm_translate_lock_patch",
+    version = "0.0.0",
+    compatibility_level = 1,
+)
```
Comment (kormide) [inline e2e/npm_translate_lock_patch/MODULE.bazel:5]: Should be able to leave out the `module` declaration here.

**Comment by kormide** [comment]
Can we rename the e2e to something like `npm_translate_lock_subdir_patch`?

File: e2e/npm_translate_lock_patch/portal/.bazelignore [hunk 3]
```diff
@@ -0,0 +1 @@
+portal/node_modules
```
Comment (kormide) [inline e2e/npm_translate_lock_patch/portal/.bazelignore:1]: Is there something meaningful about the term "portal"? If not, maybe we can just call it "subdir"?

File: e2e/npm_translate_lock_patch/MODULE.bazel [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+bazel_dep(name = "bazel_skylib", version = "1.4.1")
+bazel_dep(name = "aspect_bazel_lib", version = "1.29.2")
```
Comment (kormide) [inline e2e/npm_translate_lock_patch/MODULE.bazel:8]: Are skylib and bazel-lib needed for this e2e?

File: e2e/npm_translate_lock_patch/MODULE.bazel [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    name = "npm",
+    data = [
+        "//portal:package.json",
+        "//portal:patches/debug@4.3.4.patch",
+    ],
+    patch_args = {
+        "*": ["-p1"],
+    },
+    pnpm_lock = "//portal:pnpm-lock.yaml",
+    verify_node_modules_ignored = "//portal:.bazelignore",
```
Comment (kormide) [inline e2e/npm_translate_lock_patch/MODULE.bazel:29]: Can the `.bazelignore` file remain in the root directory?

**Comment by kormide** [comment]
Could you please add a `README.md` to the e2e to briefly explain what it tests?

**Comment by kormide** [comment]
Overall looking good. Thanks for the fix!

File: e2e/npm_translate_lock_patch/MODULE.bazel (hunk 2 above)
Comment (hunshcn) [reply e2e/npm_translate_lock_patch/MODULE.bazel]: Actually, I'm new to bazel, and I'm not sure what this means. I'll follow your advice.

File: e2e/npm_translate_lock_patch/MODULE.bazel (hunk 4 above)
Comment (hunshcn) [reply e2e/npm_translate_lock_patch/MODULE.bazel]: You're right, it seems unnecessary.

File: e2e/npm_translate_lock_patch/MODULE.bazel (hunk 5 above)
Comment (hunshcn) [reply e2e/npm_translate_lock_patch/MODULE.bazel]: of course yes

**Comment by hunshcn** [comment]
@kormide done

**Comment by hunshcn** [comment]
test failed with network, how to retry?

**Comment by kormide** [comment]
> test failed with network, how to retry?

Re-ran the tests. All pass.

**Pull request merged**